sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/XraySamplerClient.java

```diff
@@ -42,13 +87,41 @@ GetSamplingRulesResponse getSamplingRules(GetSamplingRulesRequest request) {
     }
 
     String response =
-        httpClient.fetchString(
-            "POST", getSamplingRulesEndpoint, JSON_CONTENT_TYPE, null, requestBody);
+        httpClient.fetchString("POST", endpoint, JSON_CONTENT_TYPE, null, requestBody);
 
     try {
-      return OBJECT_MAPPER.readValue(response, GetSamplingRulesResponse.class);
+      return OBJECT_MAPPER.readValue(response, responseType);
     } catch (JsonProcessingException e) {
       throw new UncheckedIOException("Failed to deserialize response.", e);
     }
   }
+
+  @SuppressWarnings("JavaUtilDate")
+  private static class FloatDateDeserializer extends StdDeserializer<Date> {
+
+    private static final long serialVersionUID = 4446058377205025341L;
+
+    private static final int AWS_DATE_MILLI_SECOND_PRECISION = 3;
+
+    private FloatDateDeserializer() {
+      super(Date.class);
+    }
+
+    @Override
+    public Date deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      return parseServiceSpecificDate(p.getText());
+    }
+
+    // Copied from AWS SDK
+    // https://github.com/aws/aws-sdk-java/blob/7b1e5b87b0bf03456df9e77716b14731adf9a7a7/aws-java-sdk-core/src/main/java/com/amazonaws/util/DateUtils.java#L239
+    /** Parses the given date string returned by the AWS service into a Date object. */
+    private static Date parseServiceSpecificDate(String dateString) {
+      try {
+        BigDecimal dateValue = new BigDecimal(dateString);
+        return new Date(dateValue.scaleByPowerOfTen(AWS_DATE_MILLI_SECOND_PRECISION).longValue());
```

Comment:
I'm as big a fan of `BigDecimal` as the next person, but isn't this just multiplying by 1000? `Long.valueOf(dateString) * 1000` seems a little simpler?

Reply:
Just for more context, even this brand new aws go sdk uses bigdecimal, so I guess it's the expected convention for aws api clients

https://github.com/aws/smithy-go/blob/main/time/time.go#L76

Reply:
yep, I think keeping consistency is fine here.